backend/python-functions/realtime_rt.py

```diff
@@ -0,0 +1,143 @@
+# Adapted from https://github.com/k-sys/covid-19/blob/master/Realtime%20R0.ipynb
+
+# Libraries required for the web version
+import pandas as pd
+import numpy as np
+from scipy import stats as sps
+from scipy.interpolate import interp1d
+
+def prepare_cases(cases, cutoff=25):
+    new_cases = cases.diff()
+
+    smoothed = new_cases.rolling(7,
+        win_type='gaussian',
+        min_periods=1,
+        center=True).mean(std=2).round()
+
+    idx_start = np.searchsorted(smoothed, cutoff)
+
+    smoothed = smoothed.iloc[idx_start:]
+    original = new_cases.loc[smoothed.index]
+
+    return original, smoothed
+
+def get_posteriors(sr, sigma=0.15):
+
+    # We create an array for every possible value of Rt
+    R_T_MAX = 12
+    r_t_range = np.linspace(0, R_T_MAX, R_T_MAX*100+1)
+
+    # Gamma is 1/serial interval
+    # https://wwwnc.cdc.gov/eid/article/26/7/20-0282_article
+    # https://www.nejm.org/doi/full/10.1056/NEJMoa2001316
+    GAMMA = 1/7
+
+    # (1) Calculate Lambda
+    lam = sr[:-1].values * np.exp(GAMMA * (r_t_range[:, None] - 1))
+
+
+    # (2) Calculate each day's likelihood
+    likelihoods = pd.DataFrame(
+        data = sps.poisson.pmf(sr[1:].values, lam),
+        index = r_t_range,
+        columns = sr.index[1:])
+
+    # (3) Create the Gaussian Matrix
+    process_matrix = sps.norm(loc=r_t_range,
+                              scale=sigma
+                             ).pdf(r_t_range[:, None])
+
+    # (3a) Normalize all rows to sum to 1
+    process_matrix /= process_matrix.sum(axis=0)
+
+    # (4) Calculate the initial prior
+    #prior0 = sps.gamma(a=4).pdf(r_t_range)
```

Comment:
Should this be removed prior to merging?

Reply:
That's my bad! This can be removed